src/main/java/org/osiam/resources/controller/GroupController.java

```diff
@@ -95,46 +72,41 @@ public void delete(@PathVariable final String id) {
             throws IOException {
-        Group group = jsonInputValidator.validateJsonGroup(request);
-        Group createdGroup = scimGroupProvisioning.replace(id, group);
-        setLocation(request, response);
-        return createdGroup;
+        Group updatedGroup = scimGroupProvisioning.replace(id, group);
+        return buildResponseWithLocation(updatedGroup, builder, HttpStatus.OK);
     }
 
     @RequestMapping(value = "/{id}", method = RequestMethod.PATCH)
-    @ResponseStatus(HttpStatus.OK)
-    @ResponseBody
-    public Group update(@PathVariable final String id, HttpServletRequest request, HttpServletResponse response)
+    public ResponseEntity<Group> update(@PathVariable final String id, @RequestBody Group group, UriComponentsBuilder builder)
             throws IOException {
-        Group group = jsonInputValidator.validateJsonGroup(request);
-        Group createdGroup = scimGroupProvisioning.update(id, group);
-        setLocation(request, response);
-        return createdGroup;
+        Group updatedGroup = scimGroupProvisioning.update(id, group);
+        return buildResponseWithLocation(updatedGroup, builder, HttpStatus.OK);
     }
 
     @RequestMapping(method = RequestMethod.GET)
     @ResponseBody
-    public SCIMSearchResult<Group> searchWithGet(HttpServletRequest request) {
-        Map<String, Object> parameterMap = requestParamHelper.getRequestParameterValues(request);
-        SCIMSearchResult<Group> scimSearchResult = scimGroupProvisioning.search((String) parameterMap.get("filter"),
-                (String) parameterMap.get("sortBy"), (String) parameterMap.get("sortOrder"),
-                (int) parameterMap.get("count"), (int) parameterMap.get("startIndex"));
-
-        return attributesRemovalHelper.removeSpecifiedAttributes(scimSearchResult, parameterMap);
+    public SCIMSearchResult<Group> searchWithGet(@RequestParam Map<String, String> requestParameters) {
+        return searchWithPost(requestParameters);
     }
 
     @RequestMapping(value = "/.search", method = RequestMethod.POST)
     @ResponseBody
-    public SCIMSearchResult<Group> searchWithPost(HttpServletRequest request) {
-        Map<String, Object> parameterMap = requestParamHelper.getRequestParameterValues(request);
-        SCIMSearchResult<Group> scimSearchResult = scimGroupProvisioning.search((String) parameterMap.get("filter"),
-                (String) parameterMap.get("sortBy"), (String) parameterMap.get("sortOrder"),
-                (int) parameterMap.get("count"), (int) parameterMap.get("startIndex"));
+    public SCIMSearchResult<Group> searchWithPost(@RequestParam Map<String, String> requestParameters) {
+        SCIMSearchResult<Group> scimSearchResult = scimGroupProvisioning.search(requestParameters.get("filter"),
+                requestParameters.get("sortBy"),
+                requestParameters.getOrDefault("sortOrder", "ascending"),
+                Integer.parseInt(requestParameters.getOrDefault("count", "100")),
```

Comment:
What about the magic number (`100`) now? Do we want to add it as a constant to `SCIMSearchResult` or some other class? As I said, it should be configurable anyway, so I don't really care. /cc @dacrome